usaspending_api/download/filestreaming/download_generation.py

```diff
@@ -305,14 +305,17 @@ def wait_for_process(process, start_time, download_job):
 
     # Let the thread run until it finishes (max MAX_VISIBILITY_TIMEOUT), with a buffer of DOWNLOAD_VISIBILITY_TIMEOUT
     sleep_count = 0
-    while process.is_alive() and (time.perf_counter() - start_time) < MAX_VISIBILITY_TIMEOUT:
+    while process.is_alive():
+        if not download_job.monthly_download and (time.perf_counter() - start_time) > MAX_VISIBILITY_TIMEOUT:
+            break
         if sleep_count < 10:
             time.sleep(WAIT_FOR_PROCESS_SLEEP / 5)
         else:
             time.sleep(WAIT_FOR_PROCESS_SLEEP)
         sleep_count += 1
 
-    if (time.perf_counter() - start_time) >= MAX_VISIBILITY_TIMEOUT or process.exitcode != 0:
+    over_time = (time.perf_counter() - start_time) >= MAX_VISIBILITY_TIMEOUT
```

Comment:
Why calculate this here when we use the same formula a few operations ago? Is accuracy that important, or are we worried about the sleeps?

Reply:
My changes were only to add a check for monthly download jobs. I am unsure of the need to recalculate. It could definitely be cleaner.